### PR TITLE
[user] Add Postgres Docker container and fetch values via GET request

### PR DESCRIPTION
**Pull request by jaylees14**
- Configure Postgres container
- Fetch values from `GET` request based on URL

Testing (will add unit tests later on, seemed less important for now):
```
~/g/s/g/T/spec-golang ❯❯❯ sh build.sh
```
```
~/g/s/g/T/spec-golang ❯❯❯ curl localhost/user/1
{"ID":1,"Name":"Andrew"}
```
```
~/g/s/g/T/spec-golang ❯❯❯ curl localhost/user/2                                                                                           
Something went wrong: User for ID 2 does not exist
```

### CODE REVIEW

File: user/user.go [hunk 1]
```diff
@@ -2,19 +2,39 @@ package main
...
+
+	userID, err := strconv.ParseInt(userIDStr, 10, 64)
+	if err != nil {
+		http.Error(w, "Invalid user ID provided", http.StatusBadRequest)
+		return
+	}
+
+	user, err := dao.GetUser(userID)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Something went wrong: %s", err.Error()), http.StatusInternalServerError)
```
Comment (c-harding) [inline user/user.go]: User doesn’t exist sounds like a 4xx (404?), not a 500

**Review by c-harding — APPROVED** [review]

File: user/user.go (hunk 1 above)
Comment (jaylees14) [reply user/user.go]: Yeah good point, I guess there's actually multiple errors that could happen here, so might be worth defining those types and switching on them

**Review by Lewiky — APPROVED** [review]
err err err.err err err

**Review by willjvsmith — APPROVED** [review]
Perfect 👌

**Pull request merged**